methods/enumerables.rb

```diff
@@ -0,0 +1,98 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |index|
+      yield self[index]
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |index|
+      yield self[index], index
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    answer = []
+    my_each do |item|
+      answer.push(item) if yield item
+    end
+    answer
+  end
+
+  def my_all?
+    return true unless block_given?
+
+    my_each do |item|
+      return false unless yield item
+    end
+    true
+  end
+
+  def my_any?
+    return true unless block_given?
+
+    my_each do |item|
+      return true unless yield item
+    end
+    false
+  end
+
+  def my_none?
+    return true unless block_given?
+
+    my_each do |item|
+      return false if yield item
+    end
+    true
+  end
+
+  def my_count(arg = nil)
+    count = 0
+    if arg
+      my_each do |item|
+        count += 1 if item == arg
+      end
+    elsif block_given?
+      my_each do |item|
+        count += 1 if (yield item) == true
+      end
+    else
+      my_each do
+        count += 1
+      end
+    end
+    count
+  end
+
+  def my_map(arg = nil)
+    return to_enum if !block_given? && !arg
+
+    my_each_with_index do |item, i|
+      self[i] = if arg
+                  arg.call(item)
+                else
+                  yield item
+                end
+    end
+    self
+  end
```

Comment:
## my_map:

- returns a new array with the results of running a given block once for every element in enum.

```ruby

      array.my_map { |num| num < 10 } === array.map { |num| num < 10 } => false

```

- does not mutate the original array